### ISSUE TITLE
Displaying players health in scoreboard

### ISSUE THREAD

**Issue by ZonK256** [garbagemule/MobArena#511]
# Feature request

**Short description**

It would be nice to have an option what to display using scoreboard. Right now there's only kill counter, while displaying hp could be more useful (especially for support classes).   

**Implementation details**

The best way would be the per-arena setting, in which it could be chosen, if scoreboard should display kills, or current xp.

**Comment by garbagemule** [comment]
Perhaps it would be better to use a different plugin to show the health of other players? Aren't there plugins out there that specifically target the scoreboard and does a bunch of stuff with them, and can't they be employed here? I don't think MobArena should do something like this if it's possible to do it with a different plugin that does it better anyway.

**Comment by ZonK256** [comment]
I have concerns if it really would be better:

- It should be displayed in sidebar for better readability,
-  It should only display and be displayed to alive players inside arena,
-  It should display players' health before targeting them (AFAIK other plugins tend to display it after entering pvp)

I'm not talking about some super fancy stuff, just a straight-forward display like scoreboard in vanilla. Wouldn't it be quite simple to add, if there's already kill counter available?

**Comment by garbagemule** [comment]
But surely there are sidebar plugins that show health of nearby players or something like that. Or maybe not? I just feel like this is a very specific thing that another plugin could actually do way better than MobArena could. I know some server owners are using plugins to display player health for healers and such. I don't know how they work or what they do, but they seem to be quite content with those solutions.

That's not to say MobArena can't do it. But calling it "straightforward" is bold. The current scoreboard implementation is very tightly coupled to kills (this is called overfitting, and it's something that only ever shows when requirements change). _Changing_ it to health wouldn't be difficult - changing it to support _both_ via a config-file option is a different matter. It can be done, but of course, if a specialized plugin is sufficient, there's no reason to invest the effort.

**Comment by ZonK256** [comment]
Well, I don't know if there's any plugin that would do so (at least not free to use plugin) and be also so tightly integrated with MA (displayed only inside arena, during play etc.). 

> Changing it to health wouldn't be difficult - changing it to support both via a config-file option is a different matter.

I'm not sure, if I explained it properly: I meant exactly 
>Changing it to health 

When I was saying about config, I meant changing it for specific arena (when it would make more sense). I mentioned support classes only to give an example "classes that would actually more appreciate health indicator than kill counter".

**Comment by garbagemule** [comment]
Yeah, so actually making it configurable as a per-arena setting is more work than simply changing from kills to health - but that won't happen because it means breaking existing functionality that there's a good chance someone is actually using and enjoying. So if this is going to be implemented, it has to be configurable.

Now, what does it mean to display health? Is it the percentage? Absolute current health? Health missing?

There's also a very logical next version of this, which you've already alluded to: healers see health, damage dealers see kills. Because why would the damage dealers be interested in health? This version seems like a bit of a nightmare to configure.

**Comment by Kryniowesegryderiusz** [comment]
I would also love to see that. Also you maybe should consider rebuilding scoreboards. It is possible to change language of everything, but unfortunatelly not scoreboard. It's also possible to display two scoreboards for ex. 10 sec each. Then you dont have to choose between health and kills. You can have both. And the most common style of displaying health is %actual_health%/%maximum_health%. And if you do not one of scoreboards you could disable it in config.
